comprl/server/player_manager.py

```diff
@@ -26,13 +28,25 @@ def authenticate(id: PlayerID, token: str) -> None:
 
     Args:
         id (PlayerID): id of the player to authenticate
-        token (str): autheticaten token
+        token (str): authenticate token
     """
-    log.debug(f"Player authenticated with id:{id} authenticated with token:{token}")
-
-    if id in _connected_players:
-        _authenticated_players[id] = True
-        matchmaking.match(id)
+    try:
+        converted_token = UUID(token)
+        user_id = user_database.verify_user(user_token=converted_token)
+        log.debug(f"Player authenticated with id:{id} authenticated with token:{token}")
+
+        if id in _connected_players:
+            _authenticated_players[id] = user_id
+            matchmaking.match(id)
+    except Exception:
+        log.debug(
+            f"Player with id {id} " f"tried to authenticate with unknown token: {token}"
+        )
+        if id in _connected_players:
+            _connected_players[id].disconnect(
+                reason="Authentication with the provided token failed"
+            )
+            remove(id)
```

Comment:
I don't get why what's the benefit of using a exception here? Imo a function `user_database.is_verified() ->bool:` should do it. 
Exceptions should only be used if there is something unordinary, not predictable by our logic happening. In e.g if there is no database file etc.

Reply:
I changed it so the database can search for any string (No more exeption from converting it to a uuid) and `user_database.verify_user()` now return an `Optional[int] `. If the returned `user_id` is `None`, the player is disconnected.